pkg/kapp/logs/pod_container_log.go

```diff
@@ -44,6 +44,30 @@ func NewPodContainerLog(
 }
 
 func (l PodContainerLog) Tail(ui ui.UI, cancelCh chan struct{}) error {
+	for {
+		linePrefix := ""
+		if len(l.opts.LinePrefix) > 0 {
+			linePrefix = l.opts.LinePrefix + " | "
+		}
+		err := l.StartTail(ui, cancelCh)
+
+		if err == io.EOF {
+			if l.opts.Follow {
+				ui.BeginLinef("Container Restarted\n")
+				time.Sleep(500 * time.Millisecond)
```

Comment:
in case container doesnt restart in 500ms, what happens?

Reply:
As container got restarted, we wait for 500ms before making a call to the `StartTail` function to obtain the stream. If it starts, then we get the stream and start outputting the data. However, if it doesn't start, then stream will be `nil` and `StartTail` will return nil. But because we are in endless `for` loop, we will continue to make calls to `StartTail` and once the container will be up, we will start streaming the logs eventually.